tutorials/using-cloudbuild-with-webhooks/index.md

```diff
@@ -0,0 +1,256 @@
+7. Go to the [Cloud Build history page](https://console.cloud.google.com/cloud-build/builds) to see your new job running.
+8. Click on the new job. \
+Instead of the “hello world” line, you should  see a json object with the content of the event payload. There are 2 fields of particular interest: **project.git\_http\_url** and **project.git\_ssh\_url**. In this tutorial we use a token for the authorisation, so we focus on the **project.git\_http\_url** field.
+9. In the Cloud Build trigger configuration, modify the inline build config once again to display only the necessary field from the event payload:
+    ```
+    steps: 
+      - name: ubuntu 
+        args: 
+          - echo 
+          - '${_GIT_REPO}' 
+    substitutions: 
+      _GIT_REPO: $(body.project.git_http_url)
+    ```
+10. Trigger the Test push event from GitLab again. In the [Cloud Build history page](https://console.cloud.google.com/cloud-build/builds) you should see the line with the GitLab http url.  
+
+### Cloning the GitLb repository
+There are 2 ways to clone a GitLab repository. The first is to use an SSH key and the other is to use tokens. In this tutorial, we focus on cloning the repository using a token. You can find the implementation using SSH [here](https://cloud.google.com/build/docs/automating-builds/build-hosted-repos-gitlab).
+
+To be able to clone the GitLab repository, you first need to create the GitLab deploy token. 
+1. In GitLab, click on **settings** and then choose **Repository**.
+2. Click on the **Expand** button next to **Deploy tokens**.
+3. Type “cloudbuild” in the **Name** field.
+4. Type “gitlab-token” in the **Username** field.
+5. Select “read\_repository” in **Scopes**.
+6. Click on the **Create deploy token** button.
+7. Copy the token value. \
+You can read more on creation of deploy tokens in GitLab [here](https://docs.gitlab.com/ee/user/project/deploy_tokens/). \
+You can use the token to clone the repository with the following command: \
+`git clone https://gitlab-token:<gitlab_token>@<git_url>`
+8. Modify the Cloud Build Trigger inline config again, replacing `[GITLAB_TOKEN]` with actual value.
+    ```
+    steps: 
+      - name: <strong>gcr.io/cloud-builders/git 
+        args: 
+          - '-c' 
+          - 'git clone https://gitlab-token:[GITLAB_TOKEN]@[${_REPO_URL](https://gitlab-token:[GITLAB_TOKEN]@${_REPO_URL)}' 
+        entrypoint: bash 
+    substitutions: 
+      _GIT_REPO: $(body.project.git_http_url) 
+      _REPO_URL: '${_GIT_REPO##https://}'
+    ```
+The parameter **project.git\_http\_url** from the event payload already contains the protocol: **https://**<git\_url>. You need to insert the token after the protocol but before the actual URL. To achieve this, you remove the protocol from the string and then build the URL again using [bash parameter extension](https://cloud.google.com/build/docs/configuring-builds/use-bash-and-bindings-in-substitutions#bash_parameter_expansions).
```

Comment:
I think you're missing a few whitespaces at the beginning of this line so that it's correctly indented to align with point 8.

Reply:
+